sip/sip-006-runtime-cost-assessment.md

```diff
@@ -746,3 +754,130 @@ types is defined as:
+Clarity contract as a `read-only` function.
+2. The user proposes the cost-assessment function by calling the
+`propose-cost` function in the **Clarity Cost Voting Contract**.
+3. Voting on the proposed function ensues via the voting functions in
+the **Clarity Cost Voting Contract**, and the function is either
+selected as a replacement for the existing cost-assessment function,
+or ignored.
+
+### Publishing a Cost-Assessment Function
+
+Cost-assessment functions to be included in proposals can be published
+in Clarity contracts. They must be written precisely as follows:
+
+1. The function should be `read-only`.
+2. The function should return a tuple with the keys `runtime`, `write_length`,
+`write_count`, `read_count`, and `read_length` (the execution cost measurement
+[categories](#measurements-for-execution-cost)).
+3. The values of the returned tuple should be Clarity expressions representing
+the asymptotic cost functions.
+3. The expression's constants and variables should be listed as arguments
+to the Clarity function.
+    * The arguments should be written in alphabetical order, and variables
+    should come before constants.
+
+Here is an example of a cost-assessment function where the runtime
+is set to `a * log(n) + b`, and all of the other parameters are constant.
+
+```Lisp
+(define-read-only (runtime_cost (n int) (a int) (b int))
+  {
+     runtime: (+ (* a (log n)) b)
+     write_length: a,
+     write_count: a,
+     read_count: a,
+     read_length: a,
+  })
+```
+
+### Making a Cost-Assessment Function Proposal
+
+Stacks holders can propose cost-assessment functions by calling the
+`propose-cost` function in the **Clarity Cost Voting Contract**. It
+can be called like so:
+
+```Lisp
+(contract-call?
+   .propose-cost
+   qualified-function-name
+   cost-function-address
+   cost-function-name)
+
+```
+
+This function will return a response containing the details of the proposal,
+and a proposal ID, if successful.
+
+Once submitted, a proposal is valid until either a supermajority miner vote
+is achieved, or until it expires, 14 days from when it was submitted.
+
+### Viewing Cost-Assessment Proposals
+
+To view pending cost-assessment function proposals, one can use the 
+following functions in the **Clarity Cost Voting Contract**:
+
+- `(get-proposal (id int))` returns a single proposal
+- `(get-proposals)` returns a list of all pending proposals
+- `(get-proposals-function (qualified-name string-ascii))` returns a list of pending proposals for a specific
+function
+
+### Voting for a Cost-Assessment Proposal
+
+**Miners**
+
+Miners can vote by creating a transaction that calls the cost assessment
+contract's `(vote (id int))` function and mining a block that includes this
+transaction. The `vote` function won't count the vote if the block wasn't
+mined by the node that signed that transaction. In other words, miners
+must **commit** their vote with their mining power.
```

Comment:
N.B. The voting contract can determine whether or not the voter is a miner, because Clarity exposes the block's miner's address (which _should_ map to the `contract-caller` variable if the miner signed the transaction using the key to the account in which the block rewards will materialize).